src/Groups/gsets.jl

```diff
@@ -820,14 +824,33 @@ julia> transitivity(mathieu_group(24))
 julia> transitivity(symmetric_group(6))
 6
 
+julia> transitivity(symmetric_group(6), 1:7)
+0
+
+julia> transitivity(symmetric_group(6), 1:5)
+ERROR: ArgumentError: the group does not act
 ```
 """
-transitivity(G::PermGroup, L::AbstractVector{Int} = 1:degree(G)) = GAP.Globals.Transitivity(G.X, GapObj(L))::Int
+function transitivity(G::PermGroup, L::AbstractVector{Int} = 1:degree(G))
+  gL = GapObj(L)
+  res = GAP.Globals.Transitivity(G.X, gL)::Int
+  res === GAP.Globals.fail && throw(ArgumentError("the group does not act"))
+  # If the result is `0` then it may be that `G` does not act on `L`,
+  # and in this case we want to throw an exception.
+  if res == 0 && length(L) > 0
+    orbs = GAP.Globals.Orbits(G.X, gL)
```

Comment:
Perhaps use `OrbitLengths` here? This might be faster in some cases?

But actually I think we can do a lot better by using our own orbit implementation here, in pure Julia (last I tried, that's much faster than GAP's orbit algorithm for permutations-on-integers). Perhaps for a future PR, and we just leave a `# TODO` comment here indicating this?

BTW, I just came upon this comment in the GAP library, in `lib/oprtperm.gi:1058`
```
##  Note that the GAP library defines transitivity also on partial orbits.
``
Ah well.... *sigh*

Reply:
Thanks.
I am going to add the '#TODO' comment.
And yes, the situation on the GAP side is subtle.